fmskill/data/altimetry.py

```diff
@@ -0,0 +1,754 @@
+
+        if satellites:
+            satellites = self.parse_satellites(satellites)
+            d["satellites"] = ",".join(satellites)
+        return d
+
+    # Create a query for satellite data as a URL pointing to the location of a CSV file with the data.
+
+    # Parameters
+    # ----------
+    # area : str
+    #     String specifying location of desired data.  The three forms allowed by the API are:
+    #         - polygon=6.811,54.993,8.009,54.993,8.009,57.154,6.811,57.154,6.811,54.993
+    #         - bbox=115,28,150,52
+    #         - lon=10.9&lat=55.9&radius=100000
+    #     A few named domains can also be used:
+    #         - GS_NorthSea, GS_BalticSea, GS_SouthChinaSea
+    # satellites : Union[List[str], str], optional
+    #     List of short or long names of satellite to include, an empty string, or the string 'sentinels' to specify
+    #         the two sentinel satellites 3a and 3b. Default: '3a'.
+    # start_time : str or datetime, optional
+    #     First date for which data is wanted, in the format '20100101' or as an empty string. If an empty string is
+    #         given, data starting from when it was first available is returned. Default: ''.
+    # end_time : str or datetime, optional
+    #     Last date for which data is wanted, in the format '20100101' or as an empty string. If an empty string is
+    #         given, data until the last time available is returned. Default: "20200101".
+    # nan_value : str, optional
+    #     Value to use to indicate bad or missing data, or an empty string to use the default (-9999). Default: ''.
+    # quality_filter : str, optional
+    #     Type of filter to apply before returning data. Currently, only the value 'dhi_combined' is allowed. If the
+    #         empty string is given, no filter is applied. Default: 'dhi_combined'.
+    # numeric : bool, optional
+    #     If True, return columns as numeric and return fewer columns in order to comply with the Met-Ocean on Demand
+    #         analysis systems. If False, all columns are returned, and string types are preserved as such.
+    #         Default: False.
+
+    def _create_query_payload(
+        self,
+        area="bbox=-11.913345,48.592117,12.411167,63.084148",
+        start_time="20200101",
+        end_time=None,
+        satellites="3a",
+        nan_value=None,
+        quality_filter=None,
+        numeric=False,
+    ) -> dict:
+        d = self._area_time_sat_payload(area, start_time, end_time, satellites)
+        if nan_value:
+            d["nodata"] = nan_value
+        if quality_filter:
+            d["qual_filters"] = quality_filter
+        if numeric:
+            d["numeric"] = numeric
+        return d
+
+    def _validate_area(self, area):
+        # polygon=6.811,54.993,8.009,54.993,8.009,57.154,6.811,57.154,6.811,54.993
+        # bbox=115.0,28.5,150.2,52.1
+        # lon=10.9&lat=55.9&radius=10.0
+        parsed = True
+        message = None
+        if area == "GS_NorthSea":
```

Comment:
This section seems very project specific :wink:

Reply:
will fix